crates/services/upgradable-executor/src/executor.rs

```diff
@@ -1111,18 +1107,22 @@ mod test {
         // If it doesn't cache the modules, the test will fail with a timeout.
         #[test]
         #[cfg(not(coverage))] // Too slow for coverage
-        #[ntest::timeout(60_000)]
         fn reuse_cached_compiled_module__native_strategy() {
             // Given
             let next_version = Executor::<Storage, DisabledRelayer>::VERSION + 1;
             let storage = storage_with_state_transition(next_version);
             let executor = Executor::native(storage, DisabledRelayer, Config::default());
             let block = valid_block(next_version);
 
+            executor.validate(&block).map(|_| ()).unwrap();
+            let start = std::time::Instant::now();
             // When
             for _ in 0..1000 {
                 let result = executor.validate(&block).map(|_| ());
 
+                if start.elapsed().as_secs() > 60 {
```

Comment:
```suggestion
                if start.elapsed().as_secs() > 1 {
```

Reply:
It doesn't get reset between each loop and before the timeout was 60 sec for the whole test, no ?

Reply:
Oh, yeah, it is true. Maybe it is better to create `start` before each `executor.validate` and check that it less than a second

Reply:
Done